Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -1431,14 +1432,14 @@ public async Task ApplySorting( params DataGridSortColumnInfo[] columns )
         await Reload();
     }
 
-    private void ResetDisplaying()
+    private async Task ResetDisplaying()
     {
         if ( Columns.IsNullOrEmpty() )
             return;
 
         foreach ( var column in Columns )
         {
-            column.Displaying = column.Displayable;
+            await column.SetDisplaying( column.Displayable );

```

Comment:
If I remember correctly `ParentDataGrid.Refresh();` just calls `StateHasChanged`. 
`StateHasChanged` should re-render when the UI thread is able to execute. I believe it will be only after the iteration is finished.